upgrade.sh

```diff
@@ -0,0 +1,351 @@
+    if [ -d ${repositoryDir} ]; then
+        rm -rf ${repositoryDir}
+    fi
+}
+
+##
+ # Download a new version
+ #
+function downloadRepository()
+{
+    echo "-- Downloading ${collectionZipUrl} to ${baseDir}"
+
+    wget ${collectionZipUrl} -O ${collectionZip} >/dev/null 2>&1
+}
+
+##
+ # Extract from compressed file
+ #
+function extractZip()
+{
+    echo "-- Extracting $project.zip..."
+
+    unzip ${collectionZip} -d ${rootDir} >/dev/null 2>&1
+
+    rm ${collectionZip}
+}
+
+##
+ # Copy classes
+ #
+function copyClasses()
+{
+    echo "-- Copying classes and contracts..."
+
+    for class in ${classes[@]}; do
+        echo "Copying ${oldNamespaceDir}.php/${class}.php..."
+
+        mkdir -p $(dirname ${newNamespaceDir}/${class})
+
+        cp ${oldNamespaceDir}/${class}.php ${newNamespaceDir}/${class}.php
+    done
+}
+
+##
+ # Move contracts
+ #
+function copyContracts()
+{
+    echo "-- Copying contracts..."
+
+    for contract in ${contracts[@]}; do
+        echo "Copying ${oldNamespaceDir}/${contract}.php..."
+
+        mkdir -p $(dirname ${newNamespaceDir}/${contract})
+
+        cp ${oldNamespaceDir}/${contract}.php ${newNamespaceDir}/${contract}.php
+    done
+}
+
+##
+ # Move traits
+ #
+function copyTraits()
+{
+    echo "-- Copying traits..."
+
+    for trait in ${traits[@]}; do
+        echo "Copying ${oldNamespaceDir}/${trait}.php..."
+
+        mkdir -p $(dirname ${newNamespaceDir}/${trait})
+
+        cp ${oldNamespaceDir}/${trait}.php ${newNamespaceDir}/${trait}.php
+    done
+}
+
+##
+ # Copy classes and contracts
+ #
+function copyStubs()
+{
+    echo "-- Copying stubs..."
+
+    for stub in ${stubs[@]}; do
+        echo "Copying ${stubsDir}/${stub} to ${rootDir}/${stub}..."
+
+        mkdir -p $(dirname ${rootDir}/${stub})
+
+        cp ${stubsDir}/${stub} ${rootDir}/${stub}
+    done
+}
+
+##
+ # Fill the alias.php file with the list of aliases
+ #
+function fillAliases()
+{
+    echo "-- Filling aliases.php..."
+
+    indent='    '
+    aliases='CARRIAGERETURN'
+
+    for contract in ${contracts[@]}; do
+        aliases="${aliases}${indent}${oldNamespace}/${contract}::class => ${newNamespace}/${contract}::class,CARRIAGERETURN"
+    done
+
+    for class in ${classes[@]}; do
+        aliases="${aliases}${indent}${oldNamespace}/${class}::class => ${newNamespace}/${class}::class,CARRIAGERETURN"
+    done
+
+    for trait in ${traits[@]}; do
+        aliases="${aliases}${indent}${oldNamespace}/${trait}::class => ${newNamespace}/${trait}::class,CARRIAGERETURN"
+    done
+
+    aliases=${aliases//\//\\\\}
+
+    sed -i "" -e "s|/\*--- ALIASES ---\*/|${aliases}|g" $aliasFile
+    sed -i "" -e "s|CARRIAGERETURN|\\${carriageReturn}|g" $aliasFile
+}
+
+##
+ # Copy tests to our tests dir
+ #
+function getCurrentVersionFromGitHub()
+{
+    echo Getting current version from $repository...
+
+    collectionVersion=$(git ls-remote $repository | grep tags/ | grep -v {} | cut -d \/ -f 3 | cut -d v -f 2 | sort --version-sort | tail -1)
```

Comment:
Crud. Just ran it and remembered this issue: Mac doesn't have `--version-sort` by default. Can we either reproduce `version-sort` here OR do a check on OS X devices to make sure they upgrade their `ls`?

https://superuser.com/questions/349897/ls-version-number-sorting-mac-os-x#349904

Reply:
I'm also using a macOS. No `ls` involved in version sorting. We are using the [sort](http://man7.org/linux/man-pages/man1/sort.1.html) command:

![image](https://user-images.githubusercontent.com/3182864/35804809-6a46b7b6-0a60-11e8-85c0-895bb2684ccc.png)

`--version-sort` should be available all around.

Reply:
@antonioribeiro Sorry, you're right that my link was about `ls` and that threw me off.

But we do have the same problem for `sort`:

![image](https://user-images.githubusercontent.com/151829/35810167-cc42a70c-0a58-11e8-88a0-5c4ce9e26561.png)

I did a bit of Googling and couldn't find much out about that error. I'll ask some other folks at Tighten if they see the same issue.

Reply:
Looks like we will have to do something like this

```
sort -t. -k 1,1n -k 2,2n -k 3,3n -k 4,4n
```

Reply:
Ah, apparently this is available on high Sierra but not Sierra. Geez.

Reply:
This is sorting it like a charm :)

```
git ls-remote https://github.com/laravel/framework.git | grep tags/ | grep -v {} | cut -d \/ -f 3 | cut -d v -f 2 | sort -t. -k 1,1n -k 2,2n -k 3,3n -k 4,4n
```

Reply:
4th part is not necessary on Laravel versioning, we can remove it, but we may have things like

```
5.3.0-RC1
```

Which, if it happens again, will not be the latest stable version

Reply:
Cleaning the results up

```
git ls-remote https://github.com/laravel/framework.git | grep tags/ | grep -v {} | cut -d \/ -f 3 | cut -d v -f 2 | grep -v RC | grep -vi beta | sort -t. -k 1,1n -k 2,2n -k 3,3n
```

Reply:
Works perfectly!

Reply:
Script updated.